packages/ui/src/lib/inputs/Input.svelte

```diff
@@ -79,8 +79,9 @@ async function onClear(): Promise<void> {
     {#if clearable}
       <button
         class="px-0.5 cursor-pointer text-[color:var(--pd-input-field-icon)] group-hover:text-[color:var(--pd-input-field-hover-icon)] group-focus-within:text-[color:var(--pd-input-field-focused-icon)]"
-        class:hidden={!value || readonly || disabled}
+        class:hidden={!value || disabled}
```

Comment:
this is not really logic, we should not be able to clear the value if we are readonly, I get why you did that, but maybe the full input should not be readonly?

Reply:
FileInput is composed of a button and an input, so I should be able to clear it if I used the button to select something

about
> we should not be able to clear the value if we are readonly

we should be not be able to clear if clearable is false, but if I say to the component : allow to clear then it's up to my responsibility to say if I want to clear it.

so to be the `|| readonly` check is something the component should not check as it's up to the consumer to decide.